pyro/contrib/examples/bart.py

```diff
@@ -0,0 +1,159 @@
+import argparse
+import csv
+import datetime
+import gzip
+import logging
+import multiprocessing
+import os
+import subprocess
+import sys
+import urllib
+
+import torch
+
+from pyro.contrib.examples.util import get_data_directory
+
+DATA = get_data_directory(__file__)
```

Comment:
I'm not sure what is the safest path to use here. I think it would be great if we followed the convention that all data was stored in `$REPO_ROOT/.data/` so that `$REPO_ROOT/examples` and `$REPO_ROOT/tutorials` could share `.data/`, but I believe that would require some refactoring.

Reply:
I agree, that will be ideal. Right now, all folders in examples have their own `.data` directory. But it is not too bad, since most of the tutorials/examples don't share that much data, and those that do can probably be organized into a single folder. Some of the examples also use `.data` to dump their outputs. I think the only organizing principle we have right now is that for an example we should only have to download the dataset once, and it shouldn't depend on where the example is run from.